ext/git/writer.go

```diff
@@ -116,3 +126,28 @@ func (m *nativeGitClient) Config(username string, email string) error {
 
 	return nil
 }
+
+// SigningConfig configures commit signing for the repository
+func (m *nativeGitClient) SigningConfig(signingkey string) error {
+	// Check if SiginingKey is a GPG key or Public SSH Key
+	keyCheck, err := regexp.MatchString(".*pub$", signingkey)
+	if err != nil {
+		return fmt.Errorf("could not validate Signing Key as GPG or Public SSH Key: %v", err)
+	}
+	if keyCheck {
+		// Setting the GPG format to ssh
+		log.Warnf("Setting GPG Format to SSH")
+		_, err = m.runCmd("config", "gpg.format", "ssh")
```

Comment:
Setting this via `git config` can easily lead to race conditions (i.e. when two processes calling the function at the same time with different signing requirements). Can you please explore how to set the signing config for the commit command on the command line? I think this function should be gone, and everything should be integrated into the `Commit` function.

Reply:
Unfortunately, it doesn't look like this is configurable via the commit cli at the moment: 

https://git-scm.com/docs/git-commit 

I have a couple of thoughts: 

1. Move it out of the commit function and somewhere it is only set once
2. Use the default `gpg` signing format and provide some instructions to mount a `~/.gitconfig` file if folks want to use SSH signing instead.

Reply:
I think setting a config is a global option, and on the command line you could do something like:

```
git -c gpg.format=ssh commit ...
```

From `git(1)`:
```
       -c <name>=<value>
           Pass a configuration parameter to the command. The value given will override values from configuration files. The <name> is expected in the same format as listed by git
           config (subkeys separated by dots).

           Note that omitting the = in git -c foo.bar ...  is allowed and sets foo.bar to the boolean true value (just like [foo]bar would in a config file). Including the equals but
           with an empty value (like git -c foo.bar= ...) sets foo.bar to the empty string which git config --type=bool will convert to false.

```

Reply:
Nice find, that works great! 

I will update the PR with that change.

Reply:
All right, let me know what you think about this change. I tested it this afternoon, and things are working as expected.